src/ConstructiveSolidGeometry/VolumePrimitives/RegularPrism.jl

```diff
@@ -78,13 +78,41 @@ end
 get_r_limits(rp::RegularPrism) = (_left_radial_interval(rp.r), _right_radial_interval(rp.r))
 get_z_limits(rp::RegularPrism) = (_left_linear_interval(rp.z), _right_linear_interval(rp.z))
 
+function get_decomposed_surfaces(rp::RegularPrism{N,T,T}) where {N, T}
+    zMin::T, zMax::T = get_z_limits(rp)
+    surfaces = AbstractSurfacePrimitive[]
+    if !isapprox(zMin, zMax, atol = geom_atol_zero(T))
+        push!(surfaces, RegularPolygon(rp, z = zMin), RegularPolygon(rp, z = zMax), RegularPrismMantle(rp, r = rp.r))
+    else
+        push!(surfaces, RegularPolygon(rp, z = zMin))
+    end
+    surfaces
+end
```

Comment:
I do not see why you are using `push!` here at all and not just returning the arrays for each of the cases.

Reply:
Changed